### DISCUSSION TITLE
a question related to teamtalk android build.

### DISCUSSION THREAD

**Discussion by HanaDadfar** [BearWare/TeamTalk5#2193] [Q&A]
@Muamalaljanahi hi, this is really my last question, and i'm asking it because in the readme it is not mentioned, Please answer this question. I now have android studio, but i don't know for building teamtalk android, which file i need to open, do i need to open build.gradle, or which file, in the readme it just says use android studio to build. Thanks, Please just answer this question. Best regards, and thanks.

**Comment by beqabeqa473** [comment]
You can read more about creating and building android projects on the
official developers.android.com page


On 3/7/24, hana ***@***.***> wrote:
> @Muamalaljanahi hi, this is really my last question, and i'm asking it
> because in the readme it is not mentioned, Please answer this question. I
> now have android studio, but i don't know for building teamtalk android,
> which file i need to open, do i need to open build.gradle, or which file, in
> the readme it just says use android studio to build. Thanks, Please just
> answer this question. Best regards, and thanks.
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/BearWare/TeamTalk5/discussions/2193
> You are receiving this because you are subscribed to this thread.
>
> Message ID: ***@***.***>


-- 
with best regards Beqa Gozalishvili
Tell: +995593454005
Email: ***@***.***
Web: https://gozaltech.org
Skype: beqabeqa473
Telegram: https://t.me/gozaltech
facebook: https://facebook.com/gozaltech
twitter: https://twitter.com/beqabeqa473
Instagram: https://instagram.com/beqa.gozalishvili

**Comment by HanaDadfar** [comment]
@beqabeqa473 i asked about teamtalk android, not any android app. In the readme, it says use android studio to build, but i don't know which file it is talking about, i mean which file should i open with android studio. Build.gradle is correct?

**Comment by beqabeqa473** [comment]
To develop for android, you should have basic understanding how
android studio works. TeamTalk-android is not different from another
android apps, and rule is more or less the same for android studio.
Open directory with a project and go ahead.

On 3/7/24, hana ***@***.***> wrote:
> @beqabeqa473 i asked about teamtalk android, not any android app. In the
> readme, it says use android studio to build, but i don't know which file it
> is talking about, i mean which file should i open with android studio.
> Build.gradle is correct?
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/BearWare/TeamTalk5/discussions/2193#discussioncomment-8708154
> You are receiving this because you were mentioned.
>
> Message ID:
> ***@***.***>


-- 
with best regards Beqa Gozalishvili
Tell: +995593454005
Email: ***@***.***
Web: https://gozaltech.org
Skype: beqabeqa473
Telegram: https://t.me/gozaltech
facebook: https://facebook.com/gozaltech
twitter: https://twitter.com/beqabeqa473
Instagram: https://instagram.com/beqa.gozalishvili

**Comment by HanaDadfar** [comment]
@beqabeqa473 i've red the android studio gide, and it also inclooded the gide for the gradle files. Please, just explane to me what file in the teamtalk android folder i should open with android studio, Then i'll do the rest my self. Please help me a little, i'm a new contributor and i whish to contribute to teamtalk. That's all please explane, Really thank you.

**Comment by HanaDadfar** [comment]
the readme is a bit unclear. It just says use android studio to build. I don't know what file should i open in android studio and continue to build. Teamtalk developers are really limited, i hope to get an answer so i will know something. Thanks to anybody who answers me @beqabeqa473 @Muamalaljanahi @CoBC @poretsky Thanks really for taking your valuable time.

**Comment by HanaDadfar** [comment]
i didn't get any answer. I asumed in open source projects, when someone has a question, the developers and contributors will help them, Especially helping a new contributor for them is important. In nvda project, we have a build instruction which explanes everything, it has more than 1000 characters, but the build instructions for teamtalk android is just around 139 characters, When removing the to try this application go to google play.
For windows, I had 1000 building errors, but i managed to build it at the end my self, if I would be waiting for the developers answers, I wouldn't be able to do that. Now, I want to build teamtalk android, and I just have a single question here to those who could build teamtalk android properly. And my question is, which file should i open in android studio? is it build.gradle? Thanks.

**Comment by HanaDadfar** [comment]
i hope someone helps me, I also wish to contribute.

**Comment by lbk2907** [comment]
Hey there, since you are managed to build using windows, may I have your
guide to build the TeamTalk on Windows platform? Thanks.

**Comment by lbk2907** [comment]
I have managed to build on Android before but I a little bit forgot how to
do that. I'll re-check that back and let you know.

**Comment by lbk2907** [comment]
What I can remember is you need to download the TeamTalkSDK for Android and
put that file someware. This is the part that I forgot. After that, if I
not mistaken, load the project in Android studio and build. This also is
the part where I forgot.

**Comment by muamel1** [comment]
You need to open the entire project in Android Studio and build from there.
In the main window, there is an Open button. Click on it and select the project folder.
Android Studio is not fully compatible with screen readers as Visual Studio is, 
so you need to be patient until you get used to it.
